src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -253,6 +262,21 @@ public void testRequiredTokenIsEmptyIntArray() throws Exception {
         }
     }
 
+    @Test
+    public void testBehaviourWithZeroChecks() throws Exception {
+        final TreeWalker treeWalker = new TreeWalker();
+        final String errMsg = "There should be no parsing or walking ASTs if there are no checks";
+        mockStatic(TreeWalker.class);
+        when(TreeWalker.parse(any(FileContents.class)))
+                .thenThrow(new IllegalStateException(errMsg));
+        try {
+            treeWalker.processFiltered(temporaryFolder.newFile("file.java"), new ArrayList<>());
+        }
+        catch (IllegalStateException ex) {
+            fail(ex.getMessage());
```

Comment:
yeah ok ! It was doing that but I thought that the convention was to use only fail. will do